app/state.go

```diff
@@ -801,7 +810,7 @@ func (u *updateStoreState) Handle(ctx *StateContext, c Controller) (State, bool)
 	installer, err := c.ReadArtifactHeaders(u.imagein)
 	if err != nil {
 		log.Errorf("Fetching Artifact headers failed: %s", err)
-		return NewUpdateCleanupState(&u.update, client.StatusFailure), false
+		return NewUpdateStatusReportState(&u.update, client.StatusFailure), false
```

Comment:
Yes, it is, because `StorePayloads` triggers the `Initialize` call of the `UpdateStorers` interface, which for Update Modules triggers the building of the directory tree. From this point it needs cleanup.

It's explained in the smaller commit message. However, since this was not the "main" part of the PR I didn't put it in the description, perhaps I should have.

Reply:
Yes, I saw the smaller commit (I also to this some times: put in the PR the main commit), but I just was't sure if the other call to clean-up should survive. With your explanation makes sense now, thanks!